src/parquet/util/cpu-info.h

```diff
@@ -54,44 +52,26 @@ class CpuInfo {
   static void VerifyCpuRequirements();
 
   /// Returns all the flags for this cpu
-  static int64_t hardware_flags() {
-    DCHECK(initialized_);
-    return hardware_flags_;
-  }
+  static int64_t hardware_flags();
 
   /// Returns whether of not the cpu supports this flag
-  inline static bool IsSupported(int64_t flag) {
-    DCHECK(initialized_);
-    return (hardware_flags_ & flag) != 0;
-  }
+  static bool IsSupported(int64_t flag);
```

Comment:
how would you suggest we keep it in the header ?
Here are the solution I can think of in order of preference:

* manual if + throw (or abort)
* Simply remove the DCHECK
* Make the logging.h a public header